File: Javascript_sol/Linked_List/Leetcode_148.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var sortList = function(head) {
    let temp = head;
    let arr = [];
    
    //Time complexity O(3N)=O(N) Space complexity O(N)
    while (temp) {
        arr.push(temp.val);
        temp = temp.next;
    };
    
    // Don't just use .sort!!!
    arr.sort((a, b) => {
  return a - b;
});
    temp = head;
    
    arr.forEach(num => {
        temp.val = num;
        temp = temp.next;
    });
    return head;
};

File: Python_sol/Linked_List/Leetcode_148.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        arr = []
        temp = head
        
        # Time complexity O(3N) = O(N) Space complexity O(N)
        while temp:
            arr.append(temp.val)
            temp = temp.next
            
        arr.sort()
        temp = head
        
        for x in arr:
            temp.val = x
            temp = temp.next
        return head